test/parallel/test-stream-finished.js

```diff
@@ -184,3 +184,21 @@ const { promisify } = require('util');
   finished(streamLike, common.mustCall);
   streamLike.emit('close');
 }
+
+{
+  const writable = new Writable({ write() {} });
+  writable.writable = false;
+  writable.destroy();
+  finished(writable, common.mustCall((err) => {
+    assert.strictEqual(err.code, 'ERR_STREAM_PREMATURE_CLOSE');
```

Comment:
Assume a writable that emitted the `'finished'` event is passed to `finished()`. Why `ERR_STREAM_PREMATURE_CLOSE`? It did not finish prematurely. I find it misleading.

Reply:
@lpinca: I agree, but that's a different issue in the code. I'm trying to split up the [other PR](https://github.com/nodejs/node/pull/29724) which was too complex to land into less controversial steps.

Reply:
I think https://github.com/nodejs/node/pull/31509 will resolve this.

Reply:
Ok.